### ISSUE TITLE
fly sync without logging in

### ISSUE THREAD

**Issue by pivotal-jamie-klassen** [concourse/concourse#2944]
tl;dr - maybe a duplicate of #1452 but @jama-pivotal can decide

## What challenge are you facing?

As we make breaking changes to the contract between fly and the API regarding login, users regularly find themselves stuck in a position where they cannot just sync their fly and they are forced to visit the web UI and click the download icons or some similar workaround.

## What would make this better?

`fly -t target sync -c <concourse url>` should just download the correct fly binary, without needing to be logged in. The download path for this binary is a stable contract, and less likely to break than login.

## Are you interested in implementing this yourself?

👍

**Comment by vito** [comment]
is the `-t target` necessary? seems like the `-c` would fully identify it

**Comment by pivotal-jamie-klassen** [comment]
@vito yeah, strictly speaking the `-t` could be optional. It might be handy if the cluster you normally target gets updated and you want to still sync (as it works currently). If you're syncing to a cluster for the first time maybe it would be helpful if adding `-t` meant that target name would be remembered in your `flyrc`.

**Comment by osis** [comment]
I'll second the notion of this being a confusing experience. CLI is giving the wrong suggestion and the only way out is to go to the Web UI or sync with another team from the same installation if one is available.

```
$ fly -t cf-notifications login -c https://wings.pivotal.io -n cf-notifications
logging in to team 'cf-notifications'

WARNING:

fly version (3.14.1) is out of sync with the target (4.2.2). to sync up, run the following:

    fly -t cf-notifications sync

error: resource not found
$ fly -t cf-notifications sync
error: unknown target: cf-notifications
```

**Comment by zoetian** [comment]
https://github.com/concourse/concourse/pull/4168
